core/src/main/java/com/graphhopper/routing/util/AbstractFlagEncoder.java

```diff
@@ -507,4 +507,9 @@ public boolean supports(Class<?> feature) {
     public boolean hasEncodedValue(String key) {
         return encodedValueLookup.hasEncodedValue(key);
     }
+
+    @Override
+    public final String toString() {
+        return name;
```

Comment:
So `FlagEncoders` get a name here. This is needed so e.g. multiple `CarFlagEncoders` can be added to `EncodingManager`, or what is the reason? For example I am wondering how can two `CarFlagEncoders` even be different from each other? If `FlagEncoder` will be refactored untils its only a collection of encoded values we certainly need a name for such collections, but so far I thought of e.g. `CarFlagEncoder` to be the encoded value collection called 'car', so why would I rename it to 'my_car'? Can you explain? 

On another note: I think `toString()` should not have any logical meaning (like object equality etc.) and only be used for debugging and logging. And since "`FlagEncoders` now have a customizable name" is kind of a story in itself maybe its useful to put it into a separate PR (this would make this PR probably much easier to comprehend what code has been changed, but maybe its also too complicated to split out).

Reply:
> This is needed so e.g. multiple CarFlagEncoders can be added to EncodingManager, or what is the reason?

I fear this is no longer necessary, will try to remove it.

> On another note: I think toString() should not have any logical meaning

Yes, definitely. Wrong choice at the beginning :(